src/bam/simulate_reads.rs

```diff
@@ -0,0 +1,181 @@
+use anyhow::Result;
+use bio::io::fasta;
+use rand::prelude::ThreadRng;
+use rand::Rng;
+use rust_htslib::bam;
+use rust_htslib::bam::Read;
+use std::collections::HashMap;
+use std::path::Path;
+use uuid::Uuid;
+
+pub fn simulate_reads<P: AsRef<Path>>(
```

Comment:
This has many args, as also the linter says. I think it is better and less error prone to use the builder pattern instead: https://idanarye.github.io/rust-typed-builder/typed_builder/derive.TypedBuilder.html.

I.e. turn the args into fields of a struct, add the derive for the TypedBuilder linked out above, and turn the function body into a method of that struct. This also helps to disentangle it a bit, allowing to separate the different logical parts of the function into separate methods.

Reply:
This is also solved when start and end are replaced by a single `Range<u64>` object.